lua/kommentary/init.lua

```diff
@@ -5,49 +5,98 @@ This module handles the initialization of the plugin.
+It will then call another function which takes care of calling the functions
+in kommentary.kommentary, which then actually comments out the range.
+You can overload this function with another function as an argument, which
+will overwrite the function that gets called at the end. The arguments
+passed to the second function are: line_number_start, line_number_end, context
 @tparam string ... Optional string indicating the mode to choose, 'single_line'
 	will operate on the current line, 'visual' will use the current visual
 	selection, otherwise it will assume a motion (Arguments will be
 	automatically passed by operatorfunc).
+@tparam ?string|function ... Optional function to call instead of the normal
+toggle_comment at the end of this function
 @treturn ?string|nil If called without arguments, it sets itself as operatorfunc
 	and returns 'g@' to be used in an expression mapping, otherwise it will
 	comment out and not return anything.
 ]]
-function M.toggle_comment(...)
+function M.go(...)
     local args = {...}
+    -- The 3 possible contexts from which this function can be called
+    local calling_context = args[1]
+    local line_number_start = nil
+    local line_number_end = nil
+    local callback_function = args[2]
+    --[[ If the second argument is not nil, it's a table containing the callback
+    function, extract that function from the table ]]
+    if type(callback_function) == "table" then
+        callback_function = callback_function[1]
+    end
     --[[ When called without any arguments (gc), return g@ (Which will be inserted
     into the mapping literaly, since the mapping is an <expr>, so you have gc
     will enter g@), before that set operatorfunc, which is the function that
     will be called after a motion, when you entered g@ before. So finally,
     typing gc will be as if you typed g@, then you can do a motion like 5j,
     now the operatorfunc gets called and has information about the motion,
     such as the range on which the motion operated. See :h operatorfunc. ]]
-    if #args <= 0 then
-        vim.api.nvim_set_option('operatorfunc', 'v:lua.kommentary.toggle_comment')
+    if calling_context == context.init then
+        vim.api.nvim_set_option('operatorfunc', 'v:lua.kommentary.go')
         return "g@"
     end
-    local modes = config.get_modes()
     --[[ Special argument passed by <Plug>KommentaryLine (gcc) to operate
     on just the current line ]]
-    if args[1] == "single_line" then
-        local row = vim.api.nvim_win_get_cursor(0)[1]
-        kommentary.toggle_comment_line(row, modes.normal)
-    elseif args[1] == "visual" then
-        local line_number_start = vim.fn.getpos('v')[2]
-        local line_number_end = vim.fn.getcurpos()[2]
-        kommentary.toggle_comment_range(line_number_start, line_number_end, modes.normal)
-    else
+    if calling_context == context.line then
+        line_number_start = vim.api.nvim_win_get_cursor(0)[1]
+	line_number_end = line_number_start
```

Comment:
```suggestion
        line_number_start = vim.api.nvim_win_get_cursor(0)[1]
        line_number_end = line_number_start
```
For some reason expandtab wasn't being set when I wrote this document, turn all tabs into spaces in a seperate commit.

Reply:
Since this affects everything, I'll fix it for everything in one go, after this PR is merged.